design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,383 @@
+
+1. Add all necessary K8s Secrets and their labels to the Secrets Provider manifest
+
+2. Run the Secrets Provider
+
+3. Run application pods
+
+### Lifecycle/Deletion
+
+The lifecycle of the Job is the length of the process. In other words, the time it takes to retrieve the Conjur secrets value and update them in the K8s Secrets. The customer can configure the Job to stay "alive" by adding **ttlSecondsAfterFinished**, with the desired about of seconds, to their Secrets Provider yaml. Because of the nature of a Job, a manual delete is not necessary.
+
+### Backwards compatibility
+
+Milestone 1 of Phase 2 will be built ontop of the current init container solution. We will delivery the same image and will not be erasing code and any current functionality.
+
+### Performance
+We will test and document how many secrets can be updated in 5 minutes on average where a secret should be either extreme long password or one vault account which is 5 vars username address port password dns
+
+### Affected Components
+- Conjur/DAP: Adding support for Job application identity granularity 
+
+## Security
+[//]: # "Are there any security issues with your solution? Even if you mentioned them somewhere in the doc it may be convenient for the security architect review to have them centralized here"
+
+## Test Plan
+|      | **Title**                                                    | **Given**                                                    | **When**                       | **Then**                                                     | **Comment** |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------ | ------------------------------------------------------------ | ----------- |
+| 1    | *Vanilla flow*, Secret Provider Job successfully updates K8s Secrets | - Conjur is running<br />- Authenticator is defined<br />- Secrets defined in Conjur and K8s Secrets are configured<br />- Service Account has correct permissions (get/update/list) <br />- Secrets Provider Job manifest is defined<br />- K8S_SECRETS_LABEL env variable is configured | Secrets Provider runs as a Job | - Secrets Provider pod authenticates and fetches Conjur secrets successfully<br />- All K8s Secrets with label(s) defined in K8S_SECRETS_LABEL are updated with most recent Conjur value <br />- App pods requesting with access to K8s secrets receive updated Conjur secret values as environment variables<br />- Secrets Provider Job terminates on completion of task |             |
+| 2    | Secret Provider Job updates all K8s Secrets                  | - Without K8S_SECRETS_LABEL env variable configured<br />- Without K8S_SECRET env variable configured in Job manifest | Secrets Provider runs as a Job | - All K8s Secrets in the namespace will be fetched and updated with most recent Conjur value<br /> |             |
```

Comment:
TODO: Two secrets providers in 1 namespace each one checks different K8s Secrets. Two secrets providers have access to same K8s Secret and read/update